### PR TITLE
Bump lnd to version v0.18.0-beta

### PR DESCRIPTION
**Pull request by guggero**
#### Pull Request Checklist
- [ ] LndServices minimum version has been updated if new lnd apis/fields are
  used.

### CODE REVIEW

File: account/watcher/watcher_test.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 
 	"github.com/btcsuite/btcd/btcec/v2"
 	secp "github.com/decred/dcrd/dcrec/secp256k1/v4"
-	gomock "github.com/golang/mock/gomock"
+	gomock "go.uber.org/mock/gomock"
```
Comment (bhandras) [inline account/watcher/watcher_test.go:12]: Interesting, so this is a fork?

**Review by bhandras — APPROVED** [review]
LGTM 🎉

File: account/watcher/watcher_test.go (hunk 1 above)
Comment (guggero) [reply account/watcher/watcher_test.go:12]: Yeah, apparently the original one was archived. Found out while attempting to fix the broken unit test by upgrading the mock.

File: go.mod [hunk 2]
```diff
@@ -181,4 +197,6 @@ replace google.golang.org/protobuf => github.com/lightninglabs/protobuf-go-hex-d
 
 replace github.com/lightninglabs/pool/auctioneerrpc => ./auctioneerrpc
 
-go 1.19
+go 1.21.4
+
+toolchain go1.22.2
```
Comment (ViktorTigerstrom) [inline go.mod:202]: Hmm shouldn't this be:
```suggestion
toolchain go1.22.3
```

Although that change should maybe be in the last commit?

**Review by ViktorTigerstrom — APPROVED** [review]
LGTM except one minor comment :rocket:! Thanks a lot

File: go.mod (hunk 2 above)
Comment (guggero) [reply go.mod]: Was generated automatically, but yes you're right. Updated.

**Pull request merged**